proto/cmp/types/v1alpha/price.proto

```diff
@@ -18,33 +18,66 @@ message PriceDetail {
-  string description = 4;
+  string description = 3;
+
+  // If not specified, the default of the boolean is false which means it is
+  // included in the price. If set to True, it means that this PriceDetail is only
+  // payable locally. For example local tourism tax that you pay at the reception of
+  // a hotel. Intended to be used in the breakdown.
+  bool locally_payable = 4;
 
   // Type of price
   string type = 5;
 
   // Type code of the price
   string type_code = 6;
 
-  // Description
-  int32 units = 7;
-
   // We can use a single PriceDetail to represent a price breakdown.
-  repeated PriceDetail breakdown = 8;
+  repeated PriceDetail breakdown = 7;
 }
 
-// ### Simple Price type
+// ### Simple Price type message Price
+//
+// Value of the price, this should be an integer converted to string.
+//
+// This field is a string intentionally. Because the currency can be a crypto
+// currency, we need a reliable way to represent big integers as most of the crypto
+// currencies have 18 decimals precision.
 //
-// Consists of a net value and a Currency type.
-message Price {
-  cmp.types.v1alpha.Currency currency = 1;
-  float net = 2;
-}
\ No newline at end of file
+// Definition of the price message: The combination of "value" and "decimals" fields
+// express always the value of the currency, not of the fraction of the currency [
+// ETH not wei, CAM and not aCAM, BTC and not Satoshi, EUR not EUR-Cents ] Be aware
+// that partners should not do rounding with crypto currencies.
+//
+// Example implementations: off-chain payment of 100 € or 100 $: value=10000
+// decimals=2 iso_currency=EUR or USD
+//
+// On-chain payment of 100.65 EURSH value=10065 decimals=2 contract_address=0x...
+//  this currency has 5 decimals on Columbus and conclusively to create the
+//  transaction value, 10065 must be divided by 10^2 = 100.65 EURSH and created in
+//  its smallest fraction by multiplying  100.65 EURSH * 10^5 => 10065000 (example
+//  conversion to bigint without losing accuracy: bigint(10065) * 10^(5-2))
+//
+// On-chain payment of 0.0065 BTC value=65 decimals=4 contract_address=0x... Using
+//  the contract address, we get the decimals decimals and the currency name or
+//  abbreviation: 8 decimals & WBTC Because we see 4 decimals specified in the
+//  message we divide 65 by 10^4 == 0.0065 WBTC (for showing in the front-end UIs)
+// 
+//  This currency has 8 decimals on-chain and conclusively to use the value of
+//  0.0065 for on-chain operations must be converted to big integer as bigint(65) *
+//  10^(8-4) == 650000
+//
+// On-chain payment of 1 nCAM value=1 decimals=9 this currency has denominator 18 on
+//  Columbus and conclusively to mint the value of 1 nCam must be divided by 10^9 =
+//  0.000000001 CAM and minted in its smallest fraction by multiplying 0.000000001 *
+//  10^18 => 1000000000 aCAM
+message Price { 
+	string value = 1; 
+	int32 decimals = 2;
+  Currency currency =3;
```

Comment:
Mixed indention - 2 spaces instead of 1 tab. ( And a space missing behind the '=' )

Reply:
done